src/core/qgspointlocator.cpp

```diff
@@ -704,13 +715,40 @@ bool QgsPointLocator::rebuildIndex( int maxFeaturesToIndex )
     }
     request.setFilterRect( rect );
   }
+
+  bool filter = false;
+  std::unique_ptr< QgsFeatureRenderer > renderer( mLayer->renderer() ? mLayer->renderer()->clone() : nullptr );
+  QgsRenderContext *ctx = nullptr;
+  if ( mContext )
+  {
+    mContext->expressionContext() << QgsExpressionContextUtils::layerScope( mLayer );
+    ctx = mContext.get();
+    if ( renderer )
+    {
+      // setup scale for scale dependent visibility (rule based)
+      renderer->startRender( *ctx, mLayer->fields() );
+      filter = renderer->capabilities() & QgsFeatureRenderer::Filter;
+      request.setSubsetOfAttributes( renderer->usedAttributes( *ctx ), mLayer->fields() );
+    }
+  }
+
   QgsFeatureIterator fi = mLayer->getFeatures( request );
   int indexedCount = 0;
+
   while ( fi.nextFeature( f ) )
   {
     if ( !f.hasGeometry() )
       continue;
 
+    if ( ctx && renderer )
+    {
+      ctx->expressionContext().setFeature( f );
+      if ( filter && !renderer->willRenderFeature( f, *ctx ) )
```

Comment:
It seems like the "filter" variable could be already checked in the "if" statement two lines above...

Reply:
right. done.